modules/sagemaker/sagemaker-studio/README.md

```diff
@@ -0,0 +1,99 @@
+# SageMaker studio Infrastructure
+
+This module contains the resources that are required to deploy the SageMaker Studio infrastructure. It defines the setup for Amazon SageMaker Studio Domain and creates SageMaker Studio User Profiles for Data Scientists and Lead Data Scientists.
+
+**NOTE** To effectively use this repository you would need to have a good understanding around AWS networking services, AWS CloudFormation and AWS CDK.
+  - [SageMaker Studio Stack](#sagemaker-studio-stack)
+  - [Getting Started](#getting-started)
+    - [Prerequisites](#prerequisites)
+    - [Module Structure](#repository-structure)
+  - [Troubleshooting](#troubleshooting)
+
+### SageMaker Studio Stack
+
+This stack handles the deployment of the following resources:
+
+1. SageMaker Studio Domain requires, along with
+2. IAM roles which would be linked to SM Studio user profiles. User Profile creating process is managed by manifests files in `manifests/shared-infra/mlops-modules.yaml`. You can simply add new entries in the list to create a new user. The user will be linked to a role depending on which group you add them to (`data_science_users` or `lead_data_science_users`).
+
+```
+  - name: data_science_users
+    value:
+    - data-scientist
+  - name: lead_data_science_users
+    value:
+    - lead-data-scientist
+```
+
+3. Default SageMaker Project Templates are also enabled on the account on the targeted region using a custom resource; the custom resource uses a lambda function, `functions/sm_studio/enable_sm_projects`, to make necessary SDK calls to both Amazon Service Catalog and Amazon SageMaker.
+
+## Inputs and outputs:
+### Required inputs:
+  - `VPC_ID`
+  - `subnet_ids`
+### Optional Inputs:
+  - `studio_domain_name`
+  - `studio_bucket_name`
+### Outputs (module metadata):
+  - `StudioDomainName`
+  - `StudioBucketName`
+  - `DataScientistRoleArn`
+  - `DataScientistRoleSSMName`
```

Comment:
When I deployed, these are the parameters I got.  The actual output doesn't seem to align with README

```
{
  "DataScientistRoleArn": "arn:aws:iam::XXXXXXXXXXXX:role/addf-mlops-sagemaker-sage-smrolesdatascientistrole-1X42E5W67SBWB",
  "LeadDataScientistRoleArn": "arn:aws:iam::XXXXXXXXXXXX:role/addf-mlops-sagemaker-sage-smrolesleaddatascientist-O5OLKFYVYA75",
  "StudioDomainId": "d-ohst6mxsuvj6",
  "StudioDomainName": "addf-mlops-sagemaker-sagemaker-sagemaker-studio-studio-domain",
  "smrolesDataScientistRoleSSMName9F0BF0DD": "/mlops/role/ds",
  "smrolesLeadDataScientistRoleSSMNameB60AB373": "/mlops/role/lead",
  "smrolesSageMakerExecutionRoleArnE3D71F83": "arn:aws:iam::XXXXXXXXXXXX:role/addf-mlops-sagemaker-sage-smrolessagemakerstudioro-EQ5A4AYXZ3XZ",
  "smrolesSageMakerExecutionRoleSSMNameC6FA1542": "/mlops/role/execution"
}
```